File: content/defends/scala/limitar-vida-variable/src/password.scala
import scala.collection.JavaConversions._

object Password {

  def main(args: Array[String]): Unit = {
    val c = System.console()
    if (c == null) {
      System.err.println("No console.")
      System.exit(1)
    }
    val username: String = c.readLine("Enter your user name: ")
    val password: Array[Char] = c.readPassword("Enter your password: ")
    val isValidUser: Boolean = verify(username, password)
    // limpiar la contraseña
    for( a <- 0 to (password.length - 1) ) {
         password(a) = '0'
      }
    if (!isValidUser) {
      throw new SecurityException("Invalid Credentials")
    }
  }

  // Validación simulada, siempre devuelve true
  private def verify(username: String, password: Array[Char]): Boolean = true

}

File: content/defends/java/limitar-vida-variable/src/password.java
class Password {

  public static void main (String args[]) throws IOException {
    Console c = System.console();

    if (c == null) {
      System.err.println("No console.");
      System.exit(1);
    }

    String username = c.readLine("Enter your user name: ");
    char[] password = c.readPassword("Enter your password: ");
    boolean isValidUser = verify(username, password);

    // limpiar la contraseña
    Arrays.fill(password,' ');

    if (!isValidUser) {
      throw new SecurityException("Invalid Credentials");
    }

  }

  // Validación simulada, siempre devuelve +true+
  private static final boolean verify(String username, char[] password) {
    return true;
  }
}
